advanced/Scripts/piholeDebug.sh

```diff
@@ -9,534 +9,1082 @@
+  # Compare the current FTL version to the remote version
+  if [[ "${FTL_VERSION}" == "$(pihole -v | awk '/FTL/ {print $6}' | cut -d ')' -f1)" ]]; then
+    # If they are the same, FTL is up-to-date
+    log_write "${TICK} ${ftl_name}: ${COL_LIGHT_GREEN}${FTL_VERSION}${COL_NC}"
   else
-    status=${status:-"Pi-hole Dashboard repository damaged."}
-    error_found=1
+    # If not, show it in yellow, signifying there is an update
+    log_write "${TICK} ${ftl_name}: ${COL_YELLOW}${FTL_VERSION}${COL_NC} (${FAQ_UPDATE_PI_HOLE})"
   fi
-  if [[ "${status}" ]]; then
-    log_echo "${status}"
-    unset status
+}
+
+# Checks the core version of the Pi-hole codebase
+check_component_versions() {
+  # Check the Web version, branch, and commit
+  compare_local_version_to_git_version "${CORE_GIT_DIRECTORY}" "Core"
+  # Check the Web version, branch, and commit
+  compare_local_version_to_git_version "${WEB_GIT_DIRECTORY}" "Web"
+  # Check the FTL version
+  check_ftl_version
+}
+
+
+get_program_version() {
+  local program_name="${1}"
+  # Create a loval variable so this function can be safely reused
+  local program_version
+  echo_current_diagnostic "${program_name} version"
+  # Evalutate the program we are checking, if it is any of the ones below, show the version
+  case "${program_name}" in
+    "lighttpd") program_version="$(${program_name} -v |& head -n1 | cut -d '/' -f2 | cut -d ' ' -f1)"
+    ;;
+    "dnsmasq") program_version="$(${program_name} -v |& head -n1 | awk '{print $3}')"
+    ;;
+    "php") program_version="$(${program_name} -v |& head -n1 | cut -d '-' -f1 | cut -d ' ' -f2)"
+    ;;
+    # If a match is not found, show an error
+    *) echo "Unrecognized program";
+  esac
+  # If the program does not have a version (the variable is empty)
+  if [[ -z "${program_version}" ]]; then
+    # Display and error
+    log_write "${CROSS} ${COL_LIGHT_RED}${program_name} version could not be detected.${COL_NC}"
+  else
+    # Otherwise, display the version
+    log_write "${INFO} ${program_version}"
   fi
+}
 
-	if light_ver=$(lighttpd -v |& head -n1 | cut -d " " -f1); then
-	  log_echo -r "${light_ver}"
-	else
-	  log_echo "lighttpd not installed."
-	  error_found=1
-	fi
-	if php_ver=$(php -v |& head -n1); then
-	  log_echo -r "${php_ver}"
-	else
-	  log_echo "PHP not installed."
-	  error_found=1
-	fi
+# These are the most critical dependencies of Pi-hole, so we check for them
+# and their versions, using the functions above.
+check_critical_program_versions() {
+  # Use the function created earlier and bundle them into one function that checks all the version numbers
+  get_program_version "dnsmasq"
+  get_program_version "lighttpd"
+  get_program_version "php"
+}
 
-	return "${error_found}"
+is_os_supported() {
+  local os_to_check="${1}"
+  # Strip just the base name of the system using sed
+  the_os=$(echo ${os_to_check} | sed 's/ .*//')
+  # If the variable is one of our supported OSes,
+  case "${the_os}" in
```

Comment:
Yes, that is correct.  Sorry for the confusion.  It' a beastly script akin to the installer....